src/models/fields/base-field-configuration.ts

```diff
@@ -1,39 +1,358 @@
 import Joi from 'joi';
-import { schema as baseFieldSchema, BaseField } from './base-field';
+import { BaseField } from './base-field';
+import {
+  schema as stringOrTranslationsSchema,
+  StringOrTranslations,
+} from '../translations';
 
-const schema = Joi.object().keys({
+const commonBaseFieldConfigurationSchema = Joi.object().keys({
   key: Joi.string().pattern(/^[a-z][a-zA-Z\d]*$/).required().example('id'),
-  name: Joi.string().required().example('ID as used in our geographic information system'),
-  inputType: Joi.string().valid('text', 'textarea', 'select', 'radio', 'switch', 'checkbox', 'file', 'files')
-    .description('The UI component to show. If not specified, it is text unless valueOptions are provided, then it is select.'),
-  valueOptions: Joi.array().items(Joi.object().keys({
-    text: Joi.string().required(),
-    value: baseFieldSchema.required().description('Will be passed through parser'),
-  })).allow(null)
-    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio'),
-  prefix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
-  suffix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  name: stringOrTranslationsSchema.required(),
+  showIf: Joi.object().keys({
+    key: Joi.string().required(),
+    value: Joi.alternatives(
+      Joi.number(),
+      Joi.string(),
+      Joi.boolean(),
+    ).required(),
+  }).description('Show this field if other field with provided key is set to provided value. Referenced field must exists already'),
   hint: Joi.string().allow('').description('As shown near the input field'),
-})
+});
+
+const prefixesMixin = {
+  prefix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  suffix: stringOrTranslationsSchema.description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+};
+
+const getBaseFieldConfigurationSchema = (
+  commonSchema: Joi.ObjectSchema,
+): Joi.AlternativesSchema => Joi.alternatives().try(
+  commonSchema.keys({
+    type: Joi.string().valid('string').default('string'),
+    defaultValue: stringOrTranslationsSchema.allow('').default(''),
+    valueOptions: Joi.array().length(0).allow(null).default(null),
+    inputType: Joi.string().valid('text', 'textarea').default('text'),
+    regex: Joi.string(),
+    ...prefixesMixin,
+  }).required(),
+  commonSchema.keys({
+    type: Joi.string().valid('string').default('string'),
+    defaultValue: stringOrTranslationsSchema.allow('').default(''),
+    valueOptions: Joi.array().items(Joi.object().keys({
+      text: stringOrTranslationsSchema.required(),
+      value: Joi.string().required().description('Will be passed through parser'),
```

Comment:
The `Will be passed through parser` comment has become irrelevant I think?

Reply:
Removed